docs/reference/snapshot-restore/restore-snapshot.asciidoc

```diff
@@ -1,175 +1,268 @@
-added to any existing data streams. For example, if only the
-`.ds-logs-2099.03.08-000003` backing index is restored from a snapshot, it is
-not automatically added to the existing `logs` data stream.
+----
+// TEST[setup:setup-snapshots]
+// TEST[s/_restore/_restore?wait_for_completion=true/]
 
 [discrete]
-=== Partial restore
+[[close-restore]]
+==== Close and restore
+
+You can restore an existing index if it's closed. However, the index in the
+snapshot must have the same number of primary shards. You can't close a data
+stream.
+
+// Only display warning in 7.x.
+// In 7.x, `action.destructive_requires_name` defaults to false.
+ifeval::["{major-version}" == "7.x"]
+WARNING: Don't use the <<indices-close,close index API>> to target the `*` or
+`.*` wildcard expressions. If you use {es}'s security features, this will close
+system indices required for authentication. To close all regular indices, use
+`*,-.*`.
+endif::[]
 
-By default, the entire restore operation will fail if one or more indices or backing indices participating in the operation don't have
-snapshots of all shards available. It can occur if some shards failed to snapshot for example. It is still possible to
-restore such indices by setting `partial` to `true`. Please note, that only successfully snapshotted shards will be
-restored in this case and all missing shards will be recreated empty.
+[source,console]
+----
+# Close an index
+POST my-index/_close
 
+# Restore the index from a snapshot
+POST _snapshot/my_repository/my_snapshot_2099.05.06/_restore 
+{
+  "indices": "my-index"
+}
+----
+// TEST[setup:setup-snapshots]
+// TEST[s/_restore/_restore?wait_for_completion=true/]
 
 [discrete]
-[[change-index-settings-during-restore]]
-=== Changing index settings during restore
+[[rename-on-restore]]
+==== Rename on restore
+
+You can rename indices and data streams you restore. Before you start, ensure
+the cluster has enough capacity to store both the existing and restored data.
 
-Use the <<restore-snapshot-api-index-settings,`index_settings`>> parameter
-to override index settings during the restore process. For example, the
-following request will restore the index `index_1` without creating any
-replicas while switching back to the default refresh interval:
+The following restore snapshot API request prepends `restored-` to the name of
+any restored index or data stream.
 
 [source,console]
------------------------------------
-POST /_snapshot/my_backup/snapshot_1/_restore
+----
+POST _snapshot/my_repository/my_snapshot_2099.05.06/_restore 
 {
-  "indices": "index_1",
-  "ignore_unavailable": true,
-  "index_settings": {
-    "index.number_of_replicas": 0
+  "indices": "my-index,logs-my_app-default",
+  "rename_pattern": "(.+)",
+  "rename_replacement": "restored-$1"
+}
+----
+// TEST[setup:setup-snapshots]
+// TEST[s/_restore/_restore?wait_for_completion=true/]
+
+When the restore operation is complete, you can delete the original indices and
```

Comment:
When the restore operation is complete, **if you choose to overwrite the current index with the restored index**, you can delete the original indices and data streams ... and reindex the restored ...

Reply:
I reworded this a bit to make it clear that the delete + reindex is optional.